modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/FilteringVectorProcess.java

```diff
@@ -0,0 +1,423 @@
+                        addDoubleValueFromFeature(f, valuesToCompare, beingFiltered);
+                } else {
+                    // is the feature in the group?
+                    if (featureComparison(groupingValues, f)) {
+                        addDoubleValueFromFeature(f, valuesToCompare, beingFiltered);
+                    } else {
+                        ((PushBackFeatureIterator) delegate).pushBack();
+                        break;
+                    }
+                }
+            }
+            next = doFiltering(beingFiltered, valuesToCompare);
+            return next != null;
+        }
+
+        private Feature doFiltering(
+                Map<Object, Feature> beingFiltered, List<Double> beingEvaluated) {
+            Object key;
+            // searches the min or max inside the values list and retrieve the feature
+            // to be returned from the Map.
+            if (beingFiltered.size() > 0 && beingEvaluated.size() > 0) {
+                if (this.aggregation.equals(Operations.MIN)) key = computeMin(beingEvaluated);
+                else key = computeMax(beingEvaluated);
+
+                return beingFiltered.get(key);
+            } else {
+                return null;
+            }
+        }
+
+        private Number computeMin(List<Double> beingEvaluated) {
+            return Collections.min(beingEvaluated);
+        }
+
+        private Number computeMax(List<Double> beingEvaluated) {
+            return Collections.max(beingEvaluated);
+        }
+
+        private boolean featureComparison(List<Object> groupingValues, Feature f) {
+            List<Object> toCompareValues = new ArrayList<>(groupingValues.size());
+            for (PropertyName p : groupByAttributes) {
+                toCompareValues.add(p.evaluate(f));
+            }
+            if (groupingValues.size() > 0 && groupingValues.equals(toCompareValues)) return true;
+            return false;
+        }
+
+        private boolean addGroupingValues(List<Object> groupingValues, Feature f) {
+            for (PropertyName p : groupByAttributes) {
+                Object result = p.evaluate(f);
+                if (result != null) groupingValues.add(p.evaluate(f));
+            }
+            if (groupingValues.isEmpty() || groupingValues.size() < groupByAttributes.size())
+                return false;
+            return true;
+        }
+
+        private void addDoubleValueFromFeature(
+                Feature f, List<Double> valuesToCompare, Map<Object, Feature> beingFiltered) {
+            Object result = this.operationAttribute.evaluate(f, Number.class);
+            if (result instanceof Number) {
+                Double value = ((Number) result).doubleValue();
+                valuesToCompare.add(value);
+                beingFiltered.put(value, f);
+
+            } else {
+                if (result != null)
+                    // not a numeric value. Throwing exception
+                    throw new ProcessException(
```

Comment:
Why in the world is that? Any Comparable will do no?

Reply:
Using Comparable now.  In case of ComplexFeatures the evaluation without providing a Context will return a Property. Added some code to check the type and eventually return the Property value as a comparable